language-support/hs/bindings/src/DA/Ledger/Jwt.hs

```diff
@@ -0,0 +1,23 @@
+-- Copyright (c) 2019 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Ledger.Jwt(
+  Jwt,
+  tryCreateFromString,
+  toString
+  ) where
+
+import Data.List.Extra (splitOn)
+
+newtype Jwt = Jwt { toString :: String }
+
+tryCreateFromString :: String -> Either String Jwt
+tryCreateFromString s = do
+  validate3parts s
+  return $ Jwt s
+
+validate3parts :: String -> Either String ()
+validate3parts s = do
+  case length (splitOn "." s) of
```

Comment:
I think this would be easier to read if you either use an `if` on the length or a pattern match on the actual list, e.g., 
```
case splitOn "." s of
  [_, _, _] -> …
  _ -> …
```
which is usually a better pattern than comparing the length anyway since it can shortcircuit once it gets passed the expected length. Admittedly, this is highly unlikely to matter here but I think the readabilty argument still applies.